transportation-data-publishing/data_tracker/signal_pms_fulcrum.py

```diff
@@ -0,0 +1,424 @@
+
+        if modified_date >= last_run_date:
+            pm_payloads.append(pgrest_record)
+
+    return pm_payloads
+
+def replace_pm_records(postgre_records, knack_pm_records, signal_records):
+    """Summary
+    
+    Args:
+        postgre_records (TYPE): Description
+        knack_pm_records (TYPE): Description
+        signal_records (TYPE): Description
+    
+    Returns:
+        TYPE: Description
+    """
+    postgre_records_df = pd.DataFrame.from_dict(postgre_records)
+    knack_pm_records_df = pd.DataFrame.from_dict(knack_pm_records.data)
+
+    pm_insert_payloads = postgre_records_df[
+        ~postgre_records_df["fulcrum_id"].isin(knack_pm_records_df["FULCRUM_ID"])
+    ].copy()
+
+    pm_update_payloads = postgre_records_df[
+        postgre_records_df["fulcrum_id"].isin(knack_pm_records_df["FULCRUM_ID"])
+    ].copy()
+
+    pm_insert_payloads["MODIFIED_DATE"] = datautil.local_timestamp()
+    pm_update_payloads["MODIFIED_DATE"] = datautil.local_timestamp()
+
+    pm_insert_payloads = map_knack_id_signal_id(signal_records, pm_insert_payloads)
+    pm_update_payloads = map_knack_id_signal_id(signal_records, pm_update_payloads)
+
+    knack_pm_records_id_df = knack_pm_records_df[["FULCRUM_ID", "id"]]
+    pm_update_payloads = pm_update_payloads.merge(
+        right=knack_pm_records_id_df,
+        left_on="fulcrum_id",
+        right_on="FULCRUM_ID",
+        how="left",
+    )
+
+    pm_insert_payloads["PM_STATUS"] = "COMPLETED"
+    pm_update_payloads["PM_STATUS"] = "COMPLETED"
+
+    pm_insert_payloads.columns = map(str.upper, pm_insert_payloads.columns)
+    pm_update_payloads.columns = map(str.upper, pm_update_payloads.columns)
+
+    pm_update_payloads = pm_update_payloads.rename(columns={"ID": "id"})
+
+    pm_insert_payloads = pm_insert_payloads.to_dict(orient="records")
+    pm_update_payloads = pm_update_payloads.to_dict(orient="records")
+
+    pm_insert_payloads = datautil.replace_keys(
+        pm_insert_payloads, knack_pm_records.field_map
+    )
+    pm_update_payloads = datautil.replace_keys(
+        pm_update_payloads, knack_pm_records.field_map
+    )
+
+    for payload in pm_insert_payloads:
+
+        insert_res = knackpy.record(
+            payload,
+            obj_key=cfg["knack_pms"]["ref_obj"],
+            api_key=KNACK_CREDENTIALS["data_tracker_test"]["api_key"],
+            app_id=KNACK_CREDENTIALS["data_tracker_test"]["app_id"],
+            method="create",
+        )
+    pdb.set_trace()
+    for payload in pm_update_payloads:
+        pdb.set_trace()
+        update_res = knackpy.record(
+            payload,
+            obj_key=cfg["knack_pms"]["ref_obj"],
+            api_key=KNACK_CREDENTIALS["data_tracker_test"]["api_key"],
+            app_id=KNACK_CREDENTIALS["data_tracker_test"]["app_id"],
+            method="update",
+        )
+
+    return len(pm_insert_payloads) + len(pm_update_payloads)
+
+
+def datetime_to_unix_timestamp(dateandtime):
+    """Summary
```

Comment:
You can remove the docstring for this very short function.

Reply:
the doc string for this function has been deleted